server/tserver/src/main/java/org/apache/accumulo/tserver/session/SessionManager.java

```diff
@@ -458,28 +458,36 @@ public List<ActiveScan> getActiveScans() {
         addActiveScan(activeScans, scanSession,
             isSingle ? ((SingleScanSession) scanSession).extent
                 : ((MultiScanSession) scanSession).threadPoolExtent,
-            ct, isSingle ? ScanType.SINGLE : ScanType.BATCH,
-            computeScanState(scanSession.getScanTask()), scanSession.scanParams, entry.getKey());
+            ct, isSingle ? ScanType.SINGLE : ScanType.BATCH, computeScanState(scanSession),
+            scanSession.scanParams, entry.getKey());
       }
     }));
 
     return activeScans;
   }
 
-  private ScanState computeScanState(ScanTask<?> scanTask) {
+  private ScanState computeScanState(ScanSession<?> session) {
     ScanState state = ScanState.RUNNING;
 
-    if (scanTask == null) {
+    if (session.getScanTask() == null) {
       state = ScanState.IDLE;
     } else {
-      switch (scanTask.getScanRunState()) {
+      switch (session.getScanTask().getScanRunState()) {
         case QUEUED:
           state = ScanState.QUEUED;
           break;
         case FINISHED:
           state = ScanState.IDLE;
           break;
         case RUNNING:
+          if (zombieCountConsumer != null) {
+            state = ScanState.ZOMBIE;
+            handleZombieScan(session);
+          } else if (session.getScanTask() == null) {
+            state = ScanState.CLEANING;
+            handleCleaningScan(session);
+          }
+          break;
```

Comment:
If a session is in the removed state where its no longer accessible to clients and it still has a thread running, then it could be considered a zombie.  The `zombieCountConsumer` instance variable is related to metrics and it being set or not does mean anything for a specific session.  The two handle methods seem to have a lot of side effects that should not happened while computing the current state of the session and could probably removed. 


```suggestion
          if(session.getState() == State.REMOVED){
            state = ScanState.ZOMBIE;
          }
          break;
```